navapbc/template-application-nextjs
# Install and configure storybook, sass, and USWDS 3.0

## Ticket

- https://wicmtdp.atlassian.net/browse/WMDP-36

## Changes
> What was added, updated, or removed in this PR.

- Install and configure storybook
- Create the first story
- Install and configure [USWDS 3.0](https://designsystem.digital.gov/) as the design system
- Modify `Dockerfile` and `docker-compose.yml` to support USWDS and storybook
- Gitignore storybook, uswds assets, and compiled css
- Prettier ignore uswds assets
- Use postinstall hook to copy uswds static assets
- Remove using CSS modules
- Updates some package dependencies
- Switch Docker base image to use alpine for increased speed

## Context for reviewers
We want to set up storybook for our front end components. This should make local development easier since this application does not have a dedicated back end. We also want to add USWDS component library to our NextJS project. Although NextJS handles the compilation of USWDS sass stylesheets automatically when we import them into our global stylesheet (`public/css/styles.css`), storybook does not. This is why we also needed to add and configure an external compiler, gulp, to successfully view USWDS styles in storybook as well. 

Heavily extends #11 to install and configure storybook, sass, and USWDS 3.0. 
- Please note that USWDS 3.0 differs significantly from USWDS 2.- in terms of configuration files. This made it difficult to directly pull from mass pfml's implementation of sass stylesheet compilation for storybook. 

## Testing
> Screenshots, GIF demos, code examples or output to help show the changes working as expected. ProTip: you can drag and drop or paste images into this textbox.

1. Follow the new directions in the `README.md` to run the Next.js app and storybook
2. Verify in your browser that both the app and storybook load with no errors (the correct font should load)
3. Make a visual change in one of the sass files 
4. Wait for sass to compile and for Next.js
5. Check in the app in your browser to verify the change is shown
